exporter/awsxrayexporter/internal/translator/segment.go

```diff
@@ -60,16 +61,173 @@ const (
+		}
+
+		return []*awsxray.Segment{segment}, nil
+	}
+
+	var hasDependency = (span.Kind() != ptrace.SpanKindServer) && (span.Kind() != ptrace.SpanKindInternal)
+	var segments []*awsxray.Segment
+
+	var serviceSegmentID = newSegmentID()
+
+	// If it is a dependency, then we need to create a subsegment
+	if hasDependency {
+		var dependencySpan ptrace.Span = ptrace.NewSpan()
+		span.CopyTo(dependencySpan)
+
+		dependencySpan.SetParentSpanID(serviceSegmentID)
+
+		dependencySubsegment, err := MakeSegment(dependencySpan, resource, indexedAttrs, indexAllAttrs, logGroupNames, skipTimestampValidation)
+
+		if err != nil {
+			return nil, err
+		}
+
+		// Set the name
+		if myAwsRemoteService, ok := span.Attributes().Get(awsRemoteService); ok {
+			dependencySubsegment.Name = awsxray.String(myAwsRemoteService.Str())
+		}
+
+		// Make this a subsegment
+		dependencySubsegment.Type = awsxray.String("subsegment")
+
+		// Remove span links from consumer spans
+		if span.Kind() == ptrace.SpanKindConsumer {
+			dependencySubsegment.Links = nil
+		}
+
+		segments = append(segments, dependencySubsegment)
+
+		// We always create a segment for the service
+		var serviceSpan ptrace.Span = ptrace.NewSpan()
+		span.CopyTo(serviceSpan)
+
+		// Set the span id to the one internally generated
+		serviceSpan.SetSpanID(serviceSegmentID)
+
+		removeAnnotations := []string{
+			"aws.remote.service",
+			"aws.remote.operation",
+			"remote.target",
+			"K8s.RemoteNamespace",
+		}
+
+		for _, v := range removeAnnotations {
+			serviceSpan.Attributes().Remove(v)
+		}
+
+		serviceSegment, err := MakeSegment(serviceSpan, resource, indexedAttrs, indexAllAttrs, logGroupNames, skipTimestampValidation)
+
+		// Set the name
+		if myAwsLocalService, ok := span.Attributes().Get(awsLocalService); ok {
+			serviceSegment.Name = awsxray.String(myAwsLocalService.Str())
+		}
+
+		// Remove the HTTP field
+		serviceSegment.HTTP = nil
+
+		// Remove AWS subsegment fields
+		serviceSegment.AWS.Operation = nil
+		serviceSegment.AWS.AccountID = nil
+		serviceSegment.AWS.RemoteRegion = nil
+		serviceSegment.AWS.RequestID = nil
+		serviceSegment.AWS.QueueURL = nil
+		serviceSegment.AWS.TableName = nil
+		serviceSegment.AWS.TableNames = nil
+
+		// Delete all metadata that does not start with 'otel.resource.'
+		for _, metaDataEntry := range serviceSegment.Metadata {
+			for key := range metaDataEntry {
+				if !strings.HasPrefix(key, "otel.resource.") {
+					delete(metaDataEntry, key)
+				}
+			}
+		}
+
+		// Make it a segment
+		serviceSegment.Type = nil
+
+		// Remove span links from non-consumer spans
+		if span.Kind() != ptrace.SpanKindConsumer {
+			serviceSegment.Links = nil
+		}
+
+		segments = append(segments, serviceSegment)
+
+		return segments, err
+	}
+
+	segment, err := MakeSegment(span, resource, indexedAttrs, indexAllAttrs, logGroupNames, skipTimestampValidation)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Make internal spans a segment
+	segment.Type = nil
```

Comment:
Comment is a bit confusing. In general, I found this method challenging to parse. Probably we could format it a bit better to make it clear what the different cases are, so the logic is easier to follow:
* Case 1: non-local root - follow generic impl
* Case 2: local service root - follow generic impl, but ensure we get a segment to represent the local service
* Case 3: local dependency root  - create segment and subsegment to represent the local and remote service